zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclUtil.java

```diff
@@ -33,4 +37,25 @@
   // Has the same effect as the ZK client using ZooDefs.Ids.CREATOR_ALL_ACL.
   public static final String SET_X509_CLIENT_ID_AS_ACL =
       ZNODE_GROUP_ACL_CONFIG_PREFIX + "setX509ClientIdAsAcl";
+  // A list of domain names that will have super user privilege, separated by ","
+  public static final String SUPER_USER_DOMAIN_NAME =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "superUserDomainName";
+  // A list of znode path prefixes, separated by ","
+  // Znode whose path starts with the defined path prefix would have open read access
+  // Meaning the znode will have (world:anyone, r) ACL
+  public static final String OPEN_READ_ACCESS_PATH_PREFIX =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "openReadAccessPathPrefix";
+  public static final String SUBJECT_ALTERNATIVE_NAME_SHORT = "SAN";
+
+  public static String[] getOpenReadAccessPathPrefixes() {
+    String openReadAccessPathPrefixesStr = System.getProperty(OPEN_READ_ACCESS_PATH_PREFIX, "");
+    return openReadAccessPathPrefixesStr == null ? new String[0] : openReadAccessPathPrefixesStr.split(",");
+  }
+
+  public static Set<String> getSuperUserDomainNames() {
+    String superUserDomainNameStr =
+        System.getProperty(ZNodeGroupAclUtil.SUPER_USER_DOMAIN_NAME, "");
+    return Arrays.stream(superUserDomainNameStr.split(","))
+        .filter(str -> str.length() > 0).collect(Collectors.toSet());
```

Comment:
Curious, is there a reason for inconsistency here? One returns an array and the other returns a set.

Reply:
The usage pattern of the result are different. For getOpenReadAccessPathPrefixes, we will iterate through each string (path prefix) from the result and see if a path contains any of the path prefix. For getSuperUserDomainNames, we want to check if any domain names returned matches with a string, so a set is suitable.